receiver/hostmetricsreceiver/internal/scraper/nfsscraper/nfs_scraper.go

```diff
@@ -33,5 +52,110 @@ func (s *nfsScraper) start(_ context.Context, _ component.Host) error {
 }
 
 func (s *nfsScraper) scrape(_ context.Context) (pmetric.Metrics, error) {
-	return pmetric.NewMetrics(), nil
+	var err error
+	var errs scrapererror.ScrapeErrors
+	now := pcommon.NewTimestampFromTime(time.Now())
+
+	s.nfsStats, err = s.getNfsStats()
```

Comment:
`nfsStats` is a field on `nfsScraper`, but then is still passed through as an argument to a scraper method where the method would have access to the field.

Either the `nfsStats` and `nfsdStats` should just be local vars and be passed in the way they are now, or the methods should access the fields on the structs rather than accepting an argument rather than the current setup of both.

My vote is to remove the fields on the struct and just use local values. These being fields on the struct would only be valuable if the values are to be maintained across scrapes, rather than being scrape-local like they appear to be here. (Additionally it means you can `nfsStats, err := s.getNfsStats()` instead of having to define `var err error` above, but this is a very minor taste improvement).